packages/gateway/src/cors.js

```diff
@@ -1,4 +1,56 @@
 /* eslint-env serviceworker */
+/**
+ * @param {Request} request
+ * @returns {Response}
+ */
+export function corsOptions(request) {
+  const headers = request.headers
+  // Make sure the necessary headers are present for this to be a valid pre-flight request
+  if (
+    headers.get('Origin') !== null &&
+    headers.get('Access-Control-Request-Method') !== null &&
+    headers.get('Access-Control-Request-Headers') !== null
+  ) {
+    // Handle CORS pre-flight request.
+    /** @type {Record<string, string>} */
+    const respHeaders = {
+      'Content-Length': '0',
+      'Access-Control-Allow-Origin': headers.get('origin') || '*',
+      'Access-Control-Allow-Methods': 'GET,POST,DELETE,OPTIONS',
+      'Access-Control-Max-Age': '86400',
+      // Allow all future content Request headers to go back to browser
+      // such as Authorization (Bearer) or X-Client-Name-Version
+      'Access-Control-Allow-Headers':
+        headers.get('Access-Control-Request-Headers') || '',
+      'Access-Control-Expose-Headers': 'Link',
+    }
+
+    return new Response(null, {
+      status: 204,
+      headers: respHeaders,
+    })
+  } else {
+    return new Response('Non CORS options request not allowed', {
```

Comment:
Really? Should be optional for the gateway no?

Reply:
yeah, I think we can make it optional. I'll remove it